index.js

```diff
@@ -0,0 +1,143 @@
+    "import/exports-last": "error",
+    "import/extensions": [
+      "error",
+      "never",
+      { js: "never", svg: "always", scss: "always" }
+    ],
+    "import/first": "error",
+    "import/named": "error",
+    "import/namespace": "error",
+    "import/newline-after-import": "error",
+    "import/no-absolute-path": "error",
+    "import/no-extraneous-dependencies": "error",
+    "import/no-mutable-exports": "error",
+    "import/no-named-as-default-member": "error",
+    "import/no-named-default": "error",
+    "import/no-self-import": "error",
+    "import/no-unresolved": "error",
+    "import/no-useless-path-segments": "error",
+    "import/no-webpack-loader-syntax": "error",
+    "import/order": ["error", { "newlines-between": "always" }],
+    "import/prefer-default-export": "off",
+    "jsx-a11y/anchor-is-valid": "error",
+    "react/boolean-prop-naming": "error",
+    "react/button-has-type": "error",
+    "react/default-props-match-prop-types": "error",
+    "react/destructuring-assignment": [
+      2,
+      "always",
+      { ignoreClassFields: true }
+    ],
+    "react/forbid-dom-props": ["error", { forbid: ["style", "id"] }],
+    "react/forbid-prop-types": "error",
+    "react/jsx-boolean-value": ["error", "never"],
+    "react/jsx-child-element-spacing": "error",
+    "react/jsx-closing-bracket-location": "error",
+    "react/jsx-closing-tag-location": "error",
+    "react/jsx-curly-brace-presence": [
+      "error",
+      { props: "never", children: "never" }
+    ],
+    "react/jsx-curly-spacing": "error",
+    "react/jsx-equals-spacing": "error",
+    "react/jsx-filename-extension": ["error", { extensions: [".js"] }],
+    "react/jsx-first-prop-new-line": "error",
+    "react/jsx-handler-names": "error",
+    "react/jsx-indent": ["error", 2],
+    "react/jsx-key": "error",
+    "react/jsx-max-depth": ["error", { max: 4 }],
+    "react/jsx-no-bind": "error",
+    "react/jsx-no-comment-textnodes": "error",
+    "react/jsx-no-duplicate-props": "error",
+    "react/jsx-no-target-blank": "error",
+    "react/jsx-one-expression-per-line": "error",
+    "react/jsx-pascal-case": "error",
+    "react/jsx-props-no-multi-spaces": "error",
+    "react/jsx-sort-default-props": "error",
+    "react/jsx-tag-spacing": ["error", { beforeClosing: "never" }],
+    "react/jsx-uses-vars": "error",
+    "react/jsx-wrap-multilines": [
+      "error",
+      {
+        declaration: "parens-new-line",
+        assignment: "parens-new-line",
+        return: "parens-new-line",
+        arrow: "parens-new-line",
+        condition: "parens-new-line",
+        logical: "parens-new-line",
+        prop: "parens-new-line"
+      }
+    ],
+    "react/forbid-foreign-prop-types": "error",
+    "react/no-access-state-in-setstate": "error",
+    "react/no-array-index-key": "error",
+    "react/no-children-prop": "error",
+    "react/no-danger": "error",
+    "react/no-deprecated": "error",
+    "react/no-did-mount-set-state": "error",
```

Comment:
Agree!

Reply:
I don't see why a setState may be done in `componentDidMount`. Besides, as you said, it makes an extra render as an initial render happens before `componentDidMount` and another one will be triggered by this change of state, which is an antipattern. You can set the initial state in `constructor`, or in `getDerivedStateFromProps` (which runs once before mounting).

@mvbattan  Could you share the link to this docs?

Even so, if there *is* in fact a use case where you'd want to do it it should have to be an exception rather than a rule. I prefer having to disable the linter for that special case.

Reply:
There you go:
https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#fetching-external-data
From these docs:
```
There is a common misconception that fetching in componentWillMount lets you avoid the first empty rendering state. In practice this was never true because React has always executed render immediately after componentWillMount. If the data is not available by the time componentWillMount fires, the first render will still show a loading state regardless of where you initiate the fetch. This is why moving the fetch to componentDidMount has no perceptible effect in the vast majority of cases.
```

Reply:
Oh, I see what you're saying now. Sure, `setState` in a callback is fine. We almost never have to di that, given that all "callbacks" for requests are in the actionCreator. For the very few cases we may have to do this, I think it's better to disable it, and have this rule to avoid `setState` outside callbacks in  `componentDidMount`. Also, we could just extract the async call to another function so this warning doesn't appear.

Unless you think there might be usual cases where this has to be done.